ed25519-dalek/tests/x25519.rs

```diff
@@ -16,16 +25,19 @@ fn ed25519_to_x25519_dh() {
     let ed25519_signing_key_a = SigningKey::from_bytes(&ed25519_secret_key_a);
     let ed25519_signing_key_b = SigningKey::from_bytes(&ed25519_secret_key_b);
 
-    let scalar_a_bytes = ed25519_signing_key_a.to_scalar_bytes();
-    let scalar_b_bytes = ed25519_signing_key_b.to_scalar_bytes();
+    let scalar_a = ed25519_signing_key_a.to_scalar();
+    let scalar_b = ed25519_signing_key_b.to_scalar();
 
+    // Compare the scalar bytes to the first 32 bytes of SHA-512(secret_key). We have to clamp and
+    // reduce the SHA-512 output because that's what the spec does before using the scalars for
+    // anything.
     assert_eq!(
-        scalar_a_bytes,
-        hex!("357c83864f2833cb427a2ef1c00a013cfdff2768d980c0a3a520f006904de90f")
+        scalar_a.to_bytes(),
+        clamp_and_reduce(&Sha512::digest(ed25519_secret_key_a)[..32]),
     );
     assert_eq!(
-        scalar_b_bytes,
-        hex!("6ebd9ed75882d52815a97585caf4790a7f6c6b3b7f821c5e259a24b02e502e11")
+        scalar_b.to_bytes(),
+        clamp_and_reduce(&Sha512::digest(ed25519_secret_key_b)[..32]),
```

Comment:
Manually verified that `Sha512::digest` matches the hex here. I removed the hex because it didn't come from any test vector. This seems like a more principled way of testing things.

Reply:
```console
$ echo "9d61b19deffd5a60ba844af492ec2cc44449c5697b326919703bac031cae7f60" | xxd -r -p | shasum -a 512 | head -c 64
357c83864f2833cb427a2ef1c00a013cfdff2768d980c0a3a520f006904de90f⏎
$ echo "4ccd089b28ff96da9db6c346ec114e0f5b8a319f35aba624da8cf6ed4fb8a6fb" | xxd -r -p | shasum -a 512 | head -c 64
6ebd9ed75882d52815a97585caf4790a7f6c6b3b7f821c5e259a24b02e502e11⏎
```